### PR TITLE
Remove index.php from URL. Credit @cdowdy #5605

### PR DESCRIPTION
**Pull request by GawainLynch**
As per discussion in https://github.com/bolt/bolt/pull/5605 this ports changes from core, in preparation for it's upcoming diet.

### CODE REVIEW

File: public/.htaccess [hunk 1]
```diff
@@ -9,7 +9,7 @@ Options -Indexes
   Options -MultiViews
 </IfModule>
 
-<FilesMatch "\.(dist|db|markdown|md|twig|yaml|yml)$">
+<FilesMatch "\.(dist|db|markdown|md|twig|yaml|yml)|(bower|composer|jsdoc|package)\.json$">
   <IfModule mod_authz_core.c>
```
Comment (CarsonF) [inline public/.htaccess]: Those aren't in the web root...

Comment (bobdenotter) [inline public/.htaccess]: Except for the files that are: `bower.json`, `package.json` and `theme.yml` are often in the `theme/` folder. 

Also, if someone [messes with their site structure](https://docs.bolt.cm/3.1/howto/troubleshooting-outside-webroot#move-the-files-outside-of-the-code-public-code-folder), this still applies.

Comment (CarsonF) [inline public/.htaccess]: Ah interesting...let me think about that.

As for people messing their structure, that's fine, but we don't need to support that here. This repo should have the minimum needed for the standard structure.

Comment (GawainLynch) [inline public/.htaccess]: > As for people messing their structure, that's fine, but we don't need to support that here. This repo should have the minimum needed for the standard structure.

Yes, I have to agree (again) that this probably isn't the place for it. One problem is that when adjusting the structure, there should be a bunch of things that are "*must be*" checked, and this is just one.

Comment (bobdenotter) [inline public/.htaccess]: The thing is, people who mess with the structure, are usually kludging their way around not being able to configure their webserver properly. These people will usually _not_ be technically proficient enough to review their `.htaccess` file for best practices. I say we leave this in, for those people.

Comment (CarsonF) [inline public/.htaccess]: Yeah. I don't like it, but that is a fair point. I'll let it rest :)

**Pull request merged**